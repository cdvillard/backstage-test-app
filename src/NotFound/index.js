// (C) Copyright 2017-2022 Hewlett Packard Enterprise Development LP

import React from 'react';
import {
  Box,
  Heading,
  Text,
  Button,
} from 'grommet';
import { useIntl } from 'react-intl';

export const NotFound = () => {
  const { formatMessage } =  useIntl();

  return (
    <Box
      margin={{ top: 'medium' }}
      align='center'
      pad='medium'
    >
      <Heading level={2}>
        {formatMessage({id: 'notFound.title'})}
      </Heading>
      <Text size='xlarge'>
        {formatMessage({id: 'notFound.description'})}
      </Text>
      <Box pad={{ vertical: 'medium' }}>
        <Button
          primary={true}
          label={formatMessage({id: 'notFound.button'})}
          onClick={() => window.history.pushState(window.history.state, '', '/')}
        />
      </Box>
      <Box align='center'>
        <Heading level='3'>
          {formatMessage({id: 'notFound.explanations.title'})}
        </Heading>
        <Box
          gap='medium'
          direction='row'
          justify='center'
          wrap={true}
        >
          <Box
            pad='small'
            width='medium'
          >
            <Heading responsive={false} margin={{ bottom: 'small', top: 'none' }} level={4}>
              {formatMessage({id: 'notFound.explanationOne.title'})}
            </Heading>
            <Text size='large'>
              {formatMessage({id: 'notFound.explanationOne.body'})}
            </Text>
          </Box>
          <Box
            pad='small'
            width='medium'
          >
            <Heading responsive={false} margin={{ bottom: 'small', top: 'none' }} level={4}>
              {formatMessage({id: 'notFound.explanationTwo.title'})}
            </Heading>
            <Text size='large'>
              {formatMessage({id: 'notFound.explanationTwo.body'})}
            </Text>
          </Box>
          <Box
            pad='small'
            width='medium'
          >
            <Heading responsive={false} margin={{ bottom: 'small', top: 'none' }} level={4}>
              {formatMessage({id: 'notFound.explanationThree.title'})}
            </Heading>
            <Text size='large'>
              {formatMessage({id: 'notFound.explanationThree.body'})}
            </Text>
          </Box>
        </Box>
      </Box>
    </Box>
  );
}
