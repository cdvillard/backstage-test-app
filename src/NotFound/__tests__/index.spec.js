// (C) Copyright 2017-2022 Hewlett Packard Enterprise Development LP

import React from 'react';
import { render } from '@testing-library/react';
import NotFound from '..';

describe('app-components/NotFound', () => {
  it('Matches snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
