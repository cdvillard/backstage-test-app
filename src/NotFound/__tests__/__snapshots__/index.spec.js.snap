// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-components/NotFound Matches snapshot 1`] = `
<div
  class="StyledBox-sc-13pk1d4-0 kqVhCV"
>
  <h2
    class="StyledHeading-sc-1rdh4aw-0 uIOfw"
  >
    notFound.title
  </h2>
  <span
    class="StyledText-sc-1sadyjn-0 fzLHNT"
  >
    notFound.description
  </span>
  <div
    class="StyledBox-sc-13pk1d4-0 keMLQv"
  >
    <button
      class="StyledButton-sc-323bzc-0 dBcUrk"
      type="button"
    >
      notFound.button
    </button>
  </div>
  <div
    class="StyledBox-sc-13pk1d4-0 hXkiLl"
  >
    <h3
      class="StyledHeading-sc-1rdh4aw-0 hfuwla"
    >
      notFound.explanations.title
    </h3>
    <div
      class="StyledBox-sc-13pk1d4-0 kjLhSK"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 hItXvo"
      >
        <h4
          class="StyledHeading-sc-1rdh4aw-0 kBKwAi"
        >
          notFound.explanationOne.title
        </h4>
        <span
          class="StyledText-sc-1sadyjn-0 flpFwE"
        >
          notFound.explanationOne.body
        </span>
      </div>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 hItXvo"
      >
        <h4
          class="StyledHeading-sc-1rdh4aw-0 kBKwAi"
        >
          notFound.explanationTwo.title
        </h4>
        <span
          class="StyledText-sc-1sadyjn-0 flpFwE"
        >
          notFound.explanationTwo.body
        </span>
      </div>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 hItXvo"
      >
        <h4
          class="StyledHeading-sc-1rdh4aw-0 kBKwAi"
        >
          notFound.explanationThree.title
        </h4>
        <span
          class="StyledText-sc-1sadyjn-0 flpFwE"
        >
          notFound.explanationThree.body
        </span>
      </div>
    </div>
  </div>
</div>
`;
